File: mobile/src/main/java/com/windscribe/mobile/splittunneling/SplitTunnelingView.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.splittunneling;

import com.windscribe.mobile.adapter.InstalledAppsAdapter;

public interface SplitTunnelingView {

    String[] getSplitRoutingModes();

    void hideTunnelSettingsLayout();

    void restartConnection();

    void setRecyclerViewAdapter(InstalledAppsAdapter mAdapter);

    void setSplitModeTextView(String mode, int textDescription);

    void setSplitRoutingModeAdapter(String[] modes, String savedMode);

    void setupToggleImage(Integer resourceId);

    void showProgress(boolean progress);

    void showTunnelSettingsLayout();

}

File: mobile/src/main/java/com/windscribe/mobile/splittunneling/SplitTunnelingView.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.mobile.splittunneling

import com.windscribe.mobile.adapter.InstalledAppsAdapter

interface SplitTunnelingView {
    val splitRoutingModes: Array<String>
    fun hideTunnelSettingsLayout()
    fun restartConnection()
    fun setRecyclerViewAdapter(mAdapter: InstalledAppsAdapter)
    fun setSplitModeTextView(mode: String, textDescription: Int)
    fun setSplitRoutingModeAdapter(
        localizeValues: Array<String>,
        selectedValue: String,
        values: Array<String>
    )

    fun setupToggleImage(resourceId: Int)
    fun showProgress(progress: Boolean)
    fun showTunnelSettingsLayout()
}